### PR TITLE
Setup dark mode with improvements

### PR DESCRIPTION
**Pull request by esilverm**
Instead of defining the colors within the global css file, we are defining the themes based on a shared file of css variables (with enhanced type checking!). We use these semantic variables alongside color primitives from before in our tailwind theme and pass through the colors as a utility in the form of a variables used with a `light` and `dark` class (not these exact names though). 

To make sure the theme is persisted properly we are using `next-theme` which stores the value as a cookie and in localhost so on there is no flickering. When the theme class appears in the document root we show the proper styles thanks to our css variables from the tailwind plugin.

### CODE REVIEW

**Comment by luketd** [comment]
Changes that were not touched by code base

StandingsTable.tsx: line 60
- If we want to keep with the theme we already have for light mode there should be something there so that the background is grey
Capture

**Comment by luketd** [comment]
Minor; scrollball when viewing players career could be greyer to match appearance of old index Capture

File: components/ColorModeSwitcher.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  useEffect(() => setMounted(true), []);
+
+  if (!mounted) return null;
+
+  return (
+    <IconButton
+      aria-label="Toggle theme"
+      className={className}
+      icon={theme === 'light' ? <MoonIcon /> : <SunIcon />}
```
Comment (luketd) [inline components/ColorModeSwitcher.tsx]: icons should be swapped as in light mode the option tno switch is the sun -> should be the moon

**Comment by luketd** [comment]
For players/[league]/schedule line 143 I believe. There should be a darker border around the box in the light mode

Capture

**Comment by luketd** [comment]
In light mode: for the options: position

the text needs to be black


Capture



In dark theme: the "Options" in Options: position, the "Options" text needs to be white

**Comment by luketd** [comment]
Optional:
From the portal suggestion box there has been a couple submissions for the lines page getting scrunched on mobile. Was going to investigate on how to make it readable on mobile after the dark theme. Not sure if you have an easy way to do it. If not Ill try to do it after

**Comment by esilverm** [comment]
> Optional: From the portal suggestion box there has been a couple submissions for the lines page getting scrunched on mobile. Was going to investigate on how to make it readable on mobile after the dark theme. Not sure if you have an easy way to do it. If not Ill try to do it after

This is something i've noticed but it is fairly inconsequential. For now let's defer it for later.

File: components/ColorModeSwitcher.tsx (hunk 1 above)
Comment (luketd) [reply components/ColorModeSwitcher.tsx]: older comment but I didnt require it. not good at github reviewing

Comment (esilverm) [reply components/ColorModeSwitcher.tsx]: you'll get the hang of it eventually

Comment (luketd) [reply components/ColorModeSwitcher.tsx]: im too used to crucible

**Pull request merged**